submissions/merowing/js-pre-oop/index.js

```diff
@@ -0,0 +1,120 @@
+import { print } from './js/lib.js';
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const data = {
+   man: {
+      id: 0,
+      hands: 2,
+      legs: 2,
+      name: 'George',
+      species: 'human',
+      gender: 'male',
+      say: 'Hello world!',
+      friends: [1, 2],
+   },
+   woman: {
+      id: 1,
+      hands: 2,
+      legs: 2,
+      name: 'Samantha',
+      gender: 'female',
+      say: 'Hi, I\'m Sam',
+      friends: [0, 3],
+   },
+   cat: {
+      id: 2,
+      legs: 4,
+      name: 'Tom',
+      gender: 'male animal',
+      species: 'cat',
+      say: 'meow-meow',
+   },
+   dog: {
+      id: 3,
+      legs: 4,
+      name: 'Allegra',
+      gender: 'female animal',
+      species: 'dog',
+      say: 'woof-woof',
+   },
+   catWoman: {
+      id: 4,
+      hands: 2,
+      legs: 2,
+      name: 'Anne',
+      gender: 'female',
+      species: 'human animal',
+      friends: [0, 1, 2],
+   }
+};
+data.catWoman.say = data.cat.say;
+data.woman.species = data.man.species;
+
+const sequences = ['species', 'name', 'gender', 'hands', 'legs', 'say', 'friends'];
+const creatures = ['man', 'woman', 'cat', 'dog', 'catWoman'];
+
+creatures.forEach(value => {
+   let creature = data[value];
+
+   let message = sequences.map(creatureProperty => {
+      let value = '';
+      
+      switch(creatureProperty) {
+         case 'name':
+            value = `<strong>${creature[creatureProperty]}</strong>`;
+            break;
```

Comment:
`switch` is an impretaive way to write code.
Consider using a dictionary (object literal) to define modifiers for properties. Then building an output could be as short as ``.map(propertyName => `<${tags[propertyName]}>${creature[propertyName]}</${tags[propertyName]}>` )``
This way the code would be more declarative, i.e. easier to read and modify.

P.S. There are "null" tags that you could benefit from.

Also consider splitting the handling of properties into a chain of transformations.
E.g. "Remove values that are 'empty', convert arrays into strings, format strings".
Having multiple simple operations vs one complex operation adds to readability and maintainability of code.